omegaconf/dictconfig.py

```diff
@@ -447,7 +447,11 @@ def _get_node(
             if throw_on_missing_key:
                 raise ConfigKeyError(f"Missing key {key}")
         elif throw_on_missing_value and value._is_missing():
-            raise MissingMandatoryValue("Missing mandatory value")
+            self._format_and_raise(
+                key=key,
+                value=None,
+                cause=MissingMandatoryValue("Missing mandatory value"),
+            )
```

Comment:
I had to change `DictConfig._get_node` (and a similar change for `ListConfig._get_node` so that the `MissingMandatoryValue` will have more information about what `key` and what `node` caused the error.